clash-lib/prims/systemverilog/Clash_Signal_Internal.json

```
@@ -3,20 +3,23 @@
     , "kind" : "Declaration"
     , "type" :
 "delay#
-  :: (HasCallStack,Undefined a) -- (ARG[0],ARG[1])
-  => Clock domain gated       -- ARG[2]
+  :: HasCallStack             -- ARG[0]
+  => Clock domain gated       -- ARG[1]
+  -> a                        -- ARG[2]
   -> Signal clk a             -- ARG[3]
   -> Signal clk a"
     , "template" :
-"// delay begin~IF ~ISGATED[2] ~THEN
-always_ff @(posedge ~ARG[2][1]) begin : ~GENSYM[~COMPNAME_delay][0]
-  if (~ARG[2][0]) begin
-    ~RESULT <= ~ARG[3];
+"// delay begin,
+~TYPO ~GENSYM[reg][0] = ~CONST[2]; ~IF ~ISGATED[1] ~THEN
+always_ff @(posedge ~ARG[1][1]) begin : ~GENSYM[~COMPNAME_delay][1]
```

```suggestion
always_ff @(posedge ~ARG[1][1]) begin : ~GENSYM[~RESULT_delay][1]
```

This makes it easier to debug longest paths